### PR TITLE
🎨 Move LIKE price in USD to VueX store

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

File: src/store/modules/getters/staticData.js [hunk 1]
```diff
@@ -40,3 +40,5 @@ export const getNFTClassIdSorter = (_, getters) => classIds => {
   });
   return sorted;
 };
+
+export const getLIKEPrice = state => state.LIKEPriceInUSD;
```
Comment (nwingt) [inline src/store/modules/getters/staticData.js:44]:
```suggestion
export const LIKEPriceInUSD = state => state.LIKEPriceInUSD;
```

File: src/store/modules/actions/staticData.js [hunk 2]
```diff
@@ -135,3 +135,24 @@ export async function lazyGetNFTOwners({ getters, dispatch }, classId) {
...
+  }
+  return price;
+}
+
+export async function lazyFetchLIKEPrice({ state, dispatch }) {
+  let price = state.likePriceInUSD;
+  if (!price) {
+    price = await dispatch('fetchLIKEPrice');
+  }
+  return price;
```
Comment (nwingt) [inline src/store/modules/actions/staticData.js:157]:
```suggestion
  if (state.likePriceInUSD) {
    return state.likePriceInUSD;
  }
  return dispatch('fetchLIKEPrice');
```

File: src/store/modules/actions/staticData.js [hunk 3]
```diff
@@ -135,3 +135,24 @@ export async function lazyGetNFTOwners({ getters, dispatch }, classId) {
...
+
+export async function fetchLIKEPrice({ commit }) {
+  let price = 0;
+  try {
+    const data = await this.$api.$get(api.getLIKEPrice());
+    price = data.likecoin.usd;
+    commit(TYPES.STATIC_SET_LIKE_PRICE_IN_USD, price);
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error('CANNOT_GET_LIKE_PRICE');
```
Comment (nwingt) [inline src/store/modules/actions/staticData.js:147]: Maybe return the previous price or rough price (if no previous price )?
https://github.com/likecoin/liker-land/blob/af8403b5fddbe408b59d9994ddad3833d99aa687/src/constant/index.js#L106

File: src/store/modules/getters/staticData.js [hunk 4]
```diff
@@ -40,3 +40,5 @@ export const getNFTClassIdSorter = (_, getters) => classIds => {
   });
   return sorted;
 };
+
+export const LIKEPriceInUSD = state => state.LIKEPriceInUSD;
```
Comment (nwingt) [inline src/store/modules/getters/staticData.js:44]:
```suggestion
export const LIKEPriceInUSD = state => state.LIKEPriceInUSD || ROUGH_LIKE_TO_USD_PRICE;
```

https://github.com/likecoin/liker-land/blob/af8403b5fddbe408b59d9994ddad3833d99aa687/src/constant/index.js#L106

**Review by williamchong — COMMENTED** [review]
Should the place where `ROUGH_LIKE_TO_USD_PRICE` is original used be using the new getter too?

File: src/pages/nft/class/_classId/index.vue [hunk 5]
```diff
@@ -177,7 +176,7 @@ export default {
                   url: this.$route.path,
                   offers: {
                     '@type': 'Offer',
-                    price: this.purchaseInfo.price * ROUGH_LIKE_TO_USD_PRICE,
+                    price: this.NFTPriceUSD,
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/pages/nft/class/_classId/index.vue:179]: How about putting NFTPriceUSD as number in nft mixin and add  formattedNFTPriceUSD as string

[the remark this answers is not in the document]
Comment (WeiJun0827) [reply src/pages/nft/class/_classId/index.vue]: Thanks for reminding!

File: src/pages/nft/class/_classId/index.vue [hunk 6]
```diff
@@ -204,12 +204,15 @@ export default {
...
     classId() {
       return this.$route.params.classId;
     },
     NFTPriceUSD() {
-      const price = this.currentPrice * this.purchaseInfo.price;
-      return `(${price.toFixed(3)} USD)`;
+      return this.LIKEPriceInUSD * this.purchaseInfo.price;
+    },
+    formattedNFTPriceUSD() {
+      return `(${this.NFTPriceUSD.toFixed(3)} USD)`;
```
Comment (nwingt) [inline src/pages/nft/class/_classId/index.vue:215]: I think it is better to move `NFTPriceUSD` and `formattedNFTPriceUSD`  to nft mixin

Comment (WeiJun0827) [reply src/pages/nft/class/_classId/index.vue:215]: Should we rename the original `nftPriceInLIKE` and `nftPriceInUSD` in NFT mixin which are fetched from strip API?

Comment (nwingt) [reply src/pages/nft/class/_classId/index.vue:215]: My suggestion,
- `nftPriceInLIKE` can be removed
- keep `nftPriceInUSD` unchanged for now
- Add `NFTPriceUSD`
- Add `formattedNFTPriceUSD `

I think it is better to have a large scale renaming in another PR

**Review by nwingt — APPROVED** [review]

**Pull request merged**